### PR TITLE
Avoid infinite loop in CreateOffer

### PR DESCRIPTION
**Pull request by jech**
Related to #1656.

### CODE REVIEW

**Comment by Sean-Der** [comment]
🔥 LGTM!

I will cherry-pick and include a test. Would also like to figure out the root cause

**Comment by jech** [comment]
>  Would also like to figure out the root cause

Absolutely.  With this patch, Galène no longer hangs, but it still fails streams when `AddSender` races with `SetRemoteDescription`.

File: peerconnection.go [hunk 1]
```diff
@@ -576,6 +577,8 @@ func (pc *PeerConnection) hasLocalDescriptionChanged(desc *SessionDescription) b
 	return false
 }
 
+var errExcessiveRetries = errors.New("excessive retries in CreateOffer")
```
[the remark this answers is not in the document]
Comment (jech) [reply peerconnection.go:580]: This should not happen in normal usage — if transceivers are being modified, then we'll only loop once or twice, while the error only appears after 128 iterations.  If this message appears, it is indicative of a bug, either in the client application or in Pion.

So perhaps we could add something like "this is probably a bug somewhere", but I feel it's overkill.

[the remark this answers is not in the document]
Comment (OrlandoCo) [reply peerconnection.go:580]: I have nerver seen this error in ion-sfu, but once I was trying to reuse transceiver I see it happen once, so I thought it was a racing issue related to mids that could not be found.

[the remark this answers is not in the document]
Comment (jech) [reply peerconnection.go:580]: The race I describe only happens when the client is slow sending its answer, so it's the kind of bug that you typically only see in production.

**Comment by Sean-Der** [comment]
Merged with 16407672bde2f1dbc7f4e9f196473ce9d09d95cd

I also added a test

**Pull request closed without merging**